ui/src/org/ensRegistrar.ts

```diff
@@ -78,15 +78,30 @@ export function formatFee(fee: ethers.BigNumber): string {
   );
 }
 
-export async function commit(
+// Expire one hour from now.
+const deadline = ethers.BigNumber.from(Math.floor(Date.now() / 1000)).add(3600);
+
+export async function getPermitSignature(
   environment: ethereum.Environment,
-  name: string,
-  salt: Uint8Array,
   fee: ethers.BigNumber
-): Promise<{
-  receipt: ethers.providers.TransactionReceipt;
+): Promise<ethers.Signature> {
+  const wallet = svelteStore.get(Wallet.store);
+  const spender = registrarAddress(environment);
+  const token = radToken();
+  return await permitSignature(wallet.signer, token, spender, fee, deadline);
```

Comment:
Could you inline this function? I don’t see a reason to split it and `permitSignature` calls `svelteStore.get(Wallet.store)` again.

Reply:
e921f99efff9d155af2a9b1db342e5a5ced8b2cb